src/mono/wasm/runtime/loader/assets.ts

```diff
@@ -220,8 +224,25 @@ export async function mono_download_assets(): Promise<void> {
 
 function prepareAssets(containedInSnapshotAssets: AssetEntryInternal[], alwaysLoadedAssets: AssetEntryInternal[]) {
     const config = loaderHelpers.config;
-    const resources = loaderHelpers.config.resources;
-    if (resources) {
+
+    // if assets exits, we will assume Net7 legacy and not process resources object
+    if (config.assets) {
+        for (const a of config.assets) {
+            const asset: AssetEntryInternal = a;
+            mono_assert(typeof asset === "object", "asset must be object");
```

Comment:
it would be good in the future to find a way to call out which specific asset is responsible for errors like this, since the list of assets could be very large